packages/libs/sdk/src/api/controllers/transactions.ts

```diff
@@ -50,22 +51,21 @@ import { TypedDocumentNode } from '@urql/core';
 import { emptyPageInfo } from '../utils/helpers';
 import { formatQueryResult } from '../utils/postQueryFormatter';
 import { isValidENSAddress } from '../utils/isValidENSAddress';
+import { ValidateInput } from '../../lib/validation/ValidateInput';
+import { balancesByWalletAddressValidator } from '../../api/types/tokens/getBalancesByWalletAddress';
 
 export class TransactionsController {
   constructor(
     private client: CustomUrqlClient,
     private defaultChain: ChainName = DEFAULT_CHAIN
   ) {}
 
+  @ValidateInput(balancesByWalletAddressValidator)
   async getByWallet(
     variables: TransactionsByWalletAddressInput
-  ): Promise<
-    TransactionsByWalletAddressResult | TransactionsByWalletENSResult
```

Comment:
These are the same result so I made it return one type (like we do in other `getByWallet` functions) for a more direct response type